sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1603,7 +1603,7 @@ def test_multitask_cv_estimators_with_sample_weight(MultiTaskEstimatorCV):
     completes smoothly as before.
     """
 
-    class CVSplitter(BaseCrossValidator, GroupsConsumerMixin):
+    class CVSplitter(GroupsConsumerMixin, BaseCrossValidator):
```

Comment:
Is modifying this test the best way to make it succeed?
Meaning that, won't it be better to have function looking at the `__metadata_request__*` of the 2 inherited classes (i.e `GroupsConsumerMixin` and `BaseCrossValidator`) and combine their `__metadata_request__*` ? 
Because right now, having `GroupsConsumerMixin` at the left or at the right of `BaseCrossValidator`, in the class inheritance, will change the value of  `CVSplitter()._get_metadata_request()` which is super confusing.

Reply:
In this case, `GroupConsumerMixin` overrides the values from `BaseCrossValidator`, so it needs to come first. Previous code was a mistake, and this fix reveals that mistake.

Reply:
Yep, usually the mixin goes always on the left.